pkg/config/config.go

```diff
@@ -38,12 +38,13 @@ type ServerConfig struct {
 	OtelCollectorPort string `mapstructure:"otel-collector-port"`
 
 	// vcs
-	VcsUsername  string `mapstructure:"vcs-username"`
-	VcsEmail     string `mapstructure:"vcs-email"`
-	VcsBaseUrl   string `mapstructure:"vcs-base-url"`
-	VcsUploadUrl string `mapstructure:"vcs-upload-url"` // github enterprise upload URL
-	VcsToken     string `mapstructure:"vcs-token"`
-	VcsType      string `mapstructure:"vcs-type"`
+	VcsUsername        string `mapstructure:"vcs-username"`
+	VcsEmail           string `mapstructure:"vcs-email"`
+	VcsBaseUrl         string `mapstructure:"vcs-base-url"`
+	VcsUploadUrl       string `mapstructure:"vcs-upload-url"` // github enterprise upload URL
+	VcsToken           string `mapstructure:"vcs-token"`
+	VcsType            string `mapstructure:"vcs-type"`
+	EnableShallowClone bool   `mapstructure:"enable-shallow-clone"`
```

Comment:
we should prefix the key with `vcs-` to make it clear what this configures.

Reply:
@djeebus I've used the `RepoShallowClone` for this instead, what do you think?

I feel `Vcs` prefixed variables are for vcs configs and this isn't one of them.